pkg/egressgateway/manager_privileged_test.go

```diff
@@ -634,3 +635,110 @@ func tryAssertEgressRules(policyMap egressmap.PolicyMap, rules []egressRule) err
 
 	return nil
 }
+
+type FakeEgressPolicyMap struct {
+	backingMap map[egressmap.EgressPolicyKey4]egressmap.EgressPolicyVal4
+}
+
+var _ egressmap.PolicyMap = (*FakeEgressPolicyMap)(nil)
+
+func (fm *FakeEgressPolicyMap) Lookup(sourceIP netip.Addr, destCIDR netip.Prefix) (*egressmap.EgressPolicyVal4, error) {
+	key := egressmap.NewEgressPolicyKey4(sourceIP, destCIDR)
+	if val, ok := fm.backingMap[key]; ok {
+		return &val, nil
+	}
+
+	return nil, errors.New("lookup: key not present in fake map")
+}
+
+func (fm *FakeEgressPolicyMap) Update(sourceIP netip.Addr, destCIDR netip.Prefix, egressIP, gatewayIP netip.Addr) error {
+	key := egressmap.NewEgressPolicyKey4(sourceIP, destCIDR)
+	val := egressmap.NewEgressPolicyVal4(egressIP, gatewayIP)
+
+	fm.backingMap[key] = val
+	return nil
+}
+
+func (fm *FakeEgressPolicyMap) Delete(sourceIP netip.Addr, destCIDR netip.Prefix) error {
+	key := egressmap.NewEgressPolicyKey4(sourceIP, destCIDR)
+	delete(fm.backingMap, key)
+	return nil
+}
+
+func (fm *FakeEgressPolicyMap) IterateWithCallback(cb egressmap.EgressPolicyIterateCallback) error {
+	for k, v := range fm.backingMap {
+		cb(&k, &v)
+	}
+	return nil
+}
+
+func BenchmarkReconcileMap(b *testing.B) {
```

Comment:
How do we know if there's performance regression happens with this test? Does it fail if some threshold is exceeded?

Reply:
I've included this to be able to add some numbers to support the change. I don't believe we have performance regression testing at the moment.